tests/test_type_conversion.py

```diff
@@ -17,6 +17,22 @@ class Foo(BaseModel):
     assert isinstance(click_type, click.types.IntParamType)
 
 
+def test_get_type_from_field_with_optional_int():
+    class Foo(BaseModel):
+        a: int | None = None
```

Comment:
Can't use `|` on Python 3.9 unfortunately...

```suggestion
        a: Optional[int] = None
```